app/src/main/java/io/capibaras/abcall/viewmodels/MainActivityViewModel.kt

```
@@ -0,0 +1,31 @@
+    var isUserLoggedIn by mutableStateOf(false)
+        private set
+
+    init {
+        checkUserSession()
+    }
+
+    private fun checkUserSession() {
+        try {
+            val token = tokenManager.getAuthToken()
+            isUserLoggedIn = !token.isNullOrEmpty()
+        } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)